### ISSUE TITLE
Add  TOP_N_VALUES/TOP_N_KEYS udfs for maps

### ISSUE THREAD

**Issue by kaikalur** [prestodb/presto#18074]
Useful function that returns just the top N keys or values from a given map.

**Comment by harshjk** [comment]
@kaikalur Please review the PR ☝️

**Comment by theWolf1198** [comment]
@kaikalur , if this is still available, can I pick it up?

**Comment by rohanpednekar** [comment]
@theWolf1198 Yes, please. I do not see @harshjk actively working on this PR; you can pick that up and address the feedback with rebase and open a new PR. We will get it reviewed. Thank you.
